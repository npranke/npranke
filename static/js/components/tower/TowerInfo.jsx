import React from 'react'

import { getSendEventHandler } from '@utils'

function TowerInfo() {
    return (
        <div className="tower-info">
            <p>
                Tower is an implementation of the Tower of Hanoi puzzle.
            </p>
            <p>
                Drag and drop disks to move the tower from origin to target.
                Larger disks can&#8217;t be placed on top of smaller disks.
                Try to move the tower in the fewest moves possible. The
                minimum number of moves is&ensp;
                <span className="math">
                    2<sup><i>n</i></sup> - 1
                </span>, where&ensp;
                <span className="math">
                    <i>n</i>
                </span>&ensp;is the number of disks in the tower.
            </p>
            <p>
                Tower is built with images of Jupiter from NASA&#8217;s Solar
                System Exploration Program. Each tower picture was taken with
                the JunoCam on the Juno spacecraft. Learn more about Jupiter,
                Juno, and other solar system missions by visiting&nbsp;
                <a
                    onClick={ getSendEventHandler(
                        'tower',
                        'navigate',
                        'NASA Solar System Exploration Program - Jupiter',
                    ) }
                    href={
                        'https://solarsystem.nasa.gov/planets/'
                        + 'jupiter/overview/'
                    }
                    target="_blank"
                    rel="noreferrer noopener external"
                >
                    NASA&#8217;s Solar System Exploration Program
                </a>.
            </p>
            <p>
                The images used in Tower are courtesy of
                NASA/JPL-Caltech. Discover more images by visiting
                the&nbsp;
                <a
                    onClick={ getSendEventHandler(
                        'tower',
                        'navigate',
                        'JPL Image Gallery',
                    ) }
                    href="https://www.jpl.nasa.gov/images"
                    target="_blank"
                    rel="noreferrer noopener external"
                >
                    JPL Image Gallery
                </a>.
            </p>
        </div>
    )
}

export default TowerInfo
