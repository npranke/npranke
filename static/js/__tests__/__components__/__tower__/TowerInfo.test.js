import { render, screen } from '@testing-library/react'

import TowerInfo from '@components/tower/TowerInfo'

describe('TowerInfo', () => {
    test('has basic description text', () => {
        render(<TowerInfo />)

        expect(
            screen.getByText(/Tower of Hanoi puzzle/),
        ).toBeInTheDocument()
    })

    test('has image credit text', () => {
        render(<TowerInfo />)

        expect(
            screen.getByText(/courtesy of NASA\/JPL-Caltech/),
        ).toBeInTheDocument()
    })

    test('has link to nasa solar system exploration program jupiter', () => {
        render(<TowerInfo />)

        expect(
            screen.getByRole(
                'link',
                { name: 'NASAâ€™s Solar System Exploration Program' },
            ),
        ).toHaveAttribute(
            'href',
            'https://solarsystem.nasa.gov/planets/jupiter/overview/',
        )
    })

    test('has link to jpl image gallery', () => {
        render(<TowerInfo />)

        expect(
            screen.getByRole('link', { name: 'JPL Image Gallery' }),
        ).toHaveAttribute('href', 'https://www.jpl.nasa.gov/images')
    })
})

describe('TowerInfo snapshot', () => {
    test('matches snapshot', () => {
        const { asFragment } = render(<TowerInfo />)

        expect(asFragment()).toMatchSnapshot()
    })
})
