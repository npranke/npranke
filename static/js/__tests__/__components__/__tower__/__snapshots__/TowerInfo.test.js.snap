// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TowerInfo snapshot matches snapshot 1`] = `
<div
  className="tower-info"
>
  <p>
    Tower is an implementation of the Tower of Hanoi puzzle.
  </p>
  <p>
    Drag and drop disks to move the tower from origin to target. Larger disks can’t be placed on top of smaller disks. Try to move the tower in the fewest moves possible. The minimum number of moves is 
    <span
      className="math"
    >
      2
      <sup>
        <i>
          n
        </i>
      </sup>
       - 1
    </span>
    , where 
    <span
      className="math"
    >
      <i>
        n
      </i>
    </span>
     is the number of disks in the tower.
  </p>
  <p>
    Tower is built with images of Jupiter from NASA’s Solar System Exploration Program. Each tower picture was taken with the JunoCam on the Juno spacecraft. Learn more about Jupiter, Juno, and other solar system missions by visiting 
    <a
      href="https://solarsystem.nasa.gov/planets/jupiter/overview/"
      onClick={[Function]}
      rel="noreferrer noopener external"
      target="_blank"
    >
      NASA’s Solar System Exploration Program
    </a>
    .
  </p>
  <p>
    The images used in Tower are courtesy of NASA/JPL-Caltech. Discover more images by visiting the 
    <a
      href="https://www.jpl.nasa.gov/images"
      onClick={[Function]}
      rel="noreferrer noopener external"
      target="_blank"
    >
      JPL Image Gallery
    </a>
    .
  </p>
</div>
`;
